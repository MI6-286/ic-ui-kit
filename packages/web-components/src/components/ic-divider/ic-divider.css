/**
  * @prop --ic-divider-vertical-height: The height of a vertical divider. The default value is `inherit`.Â 
  */
/**
  * @prop --ic-divider-horizontal-width: The width of a horizontal divider. The default value is `inherit`.
  */
/**
  * @prop --ic-divider-weight: The thickness of the divider.
  */

/* Horizontal Divider */
:host(.ic-divider-horizontal) {
  display: flex;
  width: var(--ic-divider-horizontal-width, 100%);
  align-items: center;
  column-gap: var(--ic-space-xxs);
}

hr {
  padding: 0;
  margin: 0;
  border: none;
  border-bottom: solid var(--ic-divider-weight, var(--ic-space-1px))
    var(--ic-architectural-200);
  width: inherit;
}

:host(.ic-divider-horizontal)::before,
:host(.ic-divider-horizontal)::after {
  border-left: none !important;
}

:host(.ic-divider-horizontal[label-placement="right"])::before,
:host(.ic-divider-horizontal[label-placement="center"])::before,
:host(.ic-divider-horizontal[label-placement="left"])::after,
:host(.ic-divider-horizontal[label-placement="center"])::after {
  content: "";
  border: none;
  border-bottom: solid var(--ic-divider-weight, var(--ic-space-1px))
    var(--ic-architectural-200);
  width: 100%;
}

/* Vertical Divider */
:host(.ic-divider-vertical) {
  display: flex;
  flex-direction: column;
  height: var(--ic-divider-vertical-height, inherit);
  width: max-content;
  align-items: center;
  column-gap: var(--ic-space-xxs);
  border: none;
}

:host(.ic-divider-vertical[label-placement="bottom"])::before,
:host(.ic-divider-vertical[label-placement="center"])::before,
:host(.ic-divider-vertical[label-placement="top"])::after,
:host(.ic-divider-vertical[label-placement="center"])::after {
  content: "";
  border-left: solid var(--ic-divider-weight, var(--ic-space-1px))
    var(--ic-architectural-200);
  height: 100%;
}

:host(.ic-divider-vertical) div.vertical-divider {
  display: flex;
  height: inherit;
}

/* Appearance */
:host(.ic-divider-dark) hr,
:host(.ic-divider-dark) div.vertical-divider,
:host(.ic-divider-dark[label-placement="bottom"])::before,
:host(.ic-divider-dark[label-placement="right"])::before,
:host(.ic-divider-dark[label-placement="center"])::before,
:host(.ic-divider-dark[label-placement="top"])::after,
:host(.ic-divider-dark[label-placement="left"])::after,
:host(.ic-divider-dark[label-placement="center"])::after {
  border-color: var(--ic-keyline-darken-rgb);
}
/* :host(.ic-divider-dark) hr,
:host(.ic-divider-dark.ic-divider-horizontal[label-placement="right"])::before,
:host(.ic-divider-dark.ic-divider-horizontal[label-placement="center"])::before,
:host(.ic-divider-dark.ic-divider-horizontal[label-placement="left"])::after,
:host(.ic-divider-dark.ic-divider-horizontal[label-placement="center"])::after {
  border-bottom-color: var(--ic-keyline-darken-rgb);
}

:host(.ic-divider-dark) div.vertical-divider,
:host(.ic-divider-dark.ic-divider-vertical[label-placement="bottom"])::before,
:host(.ic-divider-dark.ic-divider-vertical[label-placement="center"])::before,
:host(.ic-divider-dark.ic-divider-vertical[label-placement="top"])::after,
:host(.ic-divider-dar.ic-divider-vertical[label-placement="center"])::after {
  border-left-color: var(--ic-keyline-darken-rgb);
} */

:host(.ic-divider-light) hr,
:host(.ic-divider-light) div.vertical-divider,
:host(.ic-divider-light[label-placement="bottom"])::before,
:host(.ic-divider-light[label-placement="right"])::before,
:host(.ic-divider-light[label-placement="center"])::before,
:host(.ic-divider-light[label-placement="top"])::after,
:host(.ic-divider-light[label-placement="left"])::after,
:host(.ic-divider-light[label-placement="center"])::after {
  border-color: var(--ic-keyline-lighten-rgb);
}

:host(.ic-divider-theme) hr,
:host(.ic-divider-theme) div.vertical-divider,
:host(.ic-divider-theme[label-placement="bottom"])::before,
:host(.ic-divider-theme[label-placement="right"])::before,
:host(.ic-divider-theme[label-placement="center"])::before,
:host(.ic-divider-theme[label-placement="top"])::after,
:host(.ic-divider-theme[label-placement="left"])::after,
:host(.ic-divider-theme[label-placement="center"])::after {
  border-color: var(--ic-theme-primary);
}

/* Weight */
:host(.ic-divider-very-thick) hr,
:host(.ic-divider-very-thick[label-placement="right"])::before,
:host(.ic-divider-very-thick[label-placement="center"])::before,
:host(.ic-divider-very-thick[label-placement="left"])::after,
:host(.ic-divider-very-thick[label-placement="center"])::after {
  border-bottom-width: var(--ic-divider-weight, var(--ic-space-xs));
}

:host(.ic-divider-very-thick) .vertical-divider,
:host(.ic-divider-very-thick[label-placement="bottom"])::before,
:host(.ic-divider-very-thick[label-placement="center"])::before,
:host(.ic-divider-very-thick[label-placement="top"])::after,
:host(.ic-divider-very-thick[label-placement="center"])::after {
  border-left-width: var(--ic-divider-weight, var(--ic-space-xs));
}

:host(.ic-divider-thick) hr,
:host(.ic-divider-thick[label-placement="right"])::before,
:host(.ic-divider-thick[label-placement="center"])::before,
:host(.ic-divider-thick[label-placement="left"])::after,
:host(.ic-divider-thick[label-placement="center"])::after {
  border-bottom-width: var(--ic-divider-weight, var(--ic-space-xxs));
}

:host(.ic-divider-thick) .vertical-divider,
:host(.ic-divider-thick[label-placement="bottom"])::before,
:host(.ic-divider-thick[label-placement="center"])::before,
:host(.ic-divider-thick[label-placement="top"])::after,
:host(.ic-divider-thick[label-placement="center"])::after {
  border-left-width: var(---ic-divider-weight, var(--ic-space-xxs));
}

:host(.ic-divider-medium) hr,
:host(.ic-divider-medium[label-placement="right"])::before,
:host(.ic-divider-medium[label-placement="center"])::before,
:host(.ic-divider-medium[label-placement="left"])::after,
:host(.ic-divider-medium[label-placement="center"])::after {
  border-bottom-width: var(--ic-divider-weight, var(--ic-space-xxxs));
}

:host(.ic-divider-medium) .vertical-divider,
:host(.ic-divider-medium[label-placement="bottom"])::before,
:host(.ic-divider-medium[label-placement="center"])::before,
:host(.ic-divider-medium[label-placement="top"])::after,
:host(.ic-divider-medium[label-placement="center"])::after {
  border-left-width: var(--ic-divider-weight, var(--ic-space-xxxs));
}

:host(.ic-divider-thin) hr,
:host(.ic-divider-thin[label-placement="right"])::before,
:host(.ic-divider-thin[label-placement="center"])::before,
:host(.ic-divider-thin[label-placement="left"])::after,
:host(.ic-divider-thin[label-placement="center"])::after {
  border-bottom-width: var(--ic-divider-weight, var(--ic-space-1px));
}

:host(.ic-divider-thin) .vertical-divider,
:host(.ic-divider-thin[label-placement="bottom"])::before,
:host(.ic-divider-thin.ic-divider-vertical[label-placement="center"])::before,
:host(.ic-divider-thin[label-placement="top"])::after,
:host(.ic-divider-thin.ic-divider-vertical[label-placement="center"])::after {
  border-left-width: var(--ic-divider-weight, var(--ic-space-1px));
}

/* Style */
:host(.ic-divider-solid) .vertical-divider,
:host(.ic-divider-solid[label-placement="bottom"])::before,
:host(.ic-divider-solid[label-placement="center"])::before,
:host(.ic-divider-solid[label-placement="top"])::after,
:host(.ic-divider-solid[label-placement="center"])::after {
  border-left-style: solid;
}

:host(.ic-divider-dashed) hr,
:host(.ic-divider-dashed[label-placement="right"])::before,
:host(.ic-divider-dashed[label-placement="center"])::before,
:host(.ic-divider-dashed[label-placement="left"])::after,
:host(.ic-divider-dashed[label-placement="center"])::after {
  border-bottom-style: dashed;
}

:host(.ic-divider-dashed) .vertical-divider,
:host(.ic-divider-dashed[label-placement="bottom"])::before,
:host(.ic-divider-dashed[label-placement="center"])::before,
:host(.ic-divider-dashed[label-placement="top"])::after,
:host(.ic-divider-dashed[label-placement="center"])::after {
  border-left-style: dashed;
}

/* Label styling */
:host() .ic-divider-label p,
:host() ::slotted(ic-typography) {
  width: max-content;
}

:host(.ic-divider-default) .ic-divider-label,
:host(.ic-divider-default) ::slotted(*) {
  color: var(--ic-architectural-200);
}

:host(.ic-divider-light) .ic-divider-label,
:host(.ic-divider-light) ::slotted(*) {
  color: var(--ic-keyline-lighten-rgb);
}

:host(.ic-divider-dark) .ic-divider-label,
:host(.ic-divider-dark) ::slotted(*) {
  color: var(--ic-keyline-darken-rgb);
}

:host(.ic-divider-theme) .ic-divider-label,
:host(.ic-divider-theme) ::slotted(*) {
  color: var(--ic-theme-primary);
}
