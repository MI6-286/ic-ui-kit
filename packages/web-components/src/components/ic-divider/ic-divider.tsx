import {
  h,
  Component,
  Host,
  Listen,
  State,
  Prop,
  Element,
} from "@stencil/core";

import {
  getThemeForegroundColor,
  isPropDefined,
  slotHasContent,
} from "../../utils/helpers";

import {
  IcOrientation,
  IcTheme,
  IcThemeForeground,
  IcThemeForegroundWithTheme,
  IcThemeForegroundEnum,
} from "../../utils/types";

import {
  IcDividerLabelPlacement,
  IcDividerStyles,
  IcDividerWeights,
} from "./ic-divider.types";

@Component({
  tag: "ic-divider",
  styleUrl: "ic-divider.css",
  scoped: true,
})
export class Divider {
  @Element() el: HTMLIcDividerElement;

  @State() foregroundColor: IcThemeForeground = getThemeForegroundColor();
  /**
   * The appearance of the divider.
   */
  @Prop() appearance: IcThemeForegroundWithTheme = "default";
  /**
   * The line style of the divider.
   */
  @Prop() borderStyle: IcDividerStyles = "solid";
  /**
   * The thickness of the divider.
   */
  @Prop() weight: IcDividerWeights = "thin";
  /**
   * The orientation of the divider.
   */
  @Prop() orientation: IcOrientation = "horizontal";
  /**
   * The position the label is placed on the divider. `Left` and `right` placement is only applicable when orientation is set to `horizontal`. `Top` and `bottom` placement is only applicable when orientation is set to `vertical`. `Center` placement is applicable for both orientations.
   */
  @Prop() labelPlacement?: IcDividerLabelPlacement = "center";
  /**
   * The label for the divider.
   */
  @Prop() label?: string;

  @Listen("themeChange", { target: "document" })
  themeChangeHandler(ev: CustomEvent): void {
    const theme: IcTheme = ev.detail;
    this.foregroundColor = theme.mode;
  }

  render() {
    const {
      orientation,
      borderStyle,
      weight,
      label,
      labelPlacement,
      foregroundColor,
    } = this;

    let { appearance } = this;

    const invalidPropCombination = (
      placement: IcDividerLabelPlacement
    ): void => {
      console.error(
        `${
          placement.charAt(0).toUpperCase() + placement.slice(1)
        } label placement is not applicable for a ${orientation} ic-divider (web-components) / IcDivider (react)`
      );
    };

    if (
      ((labelPlacement === "left" || labelPlacement === "right") &&
        orientation === "vertical") ||
      ((labelPlacement === "top" || labelPlacement === "bottom") &&
        orientation === "horizontal")
    ) {
      invalidPropCombination(labelPlacement);
    }

    const getTypographyVariant = (weight: string) => {
      switch (weight) {
        case "very-thick":
          return "h4";
        case "thick":
          return "subtitle-large";
        default:
          return "label";
      }
    };

    const renderLabel = () => {
      if (slotHasContent(this.el, "label")) {
        return <slot name="label"></slot>;
      } else if (isPropDefined(label)) {
        return (
          <ic-typography
            class="ic-divider-label"
            variant={getTypographyVariant(weight)}
          >
            <p>{label}</p>
          </ic-typography>
        );
      }
      return null;
    };

    if (
      this.el.parentElement.tagName === "IC-SIDE-NAVIGATION" &&
      appearance !== "theme"
    ) {
      if (foregroundColor === IcThemeForegroundEnum.Light) {
        appearance = "light";
      } else if (foregroundColor === IcThemeForegroundEnum.Dark) {
        appearance = "dark";
      }
    }

    return (
      <Host
        class={{
          [`ic-divider-${appearance}`]: true,
          [`ic-divider-${orientation}`]: true,
          [`ic-divider-weight`]: true,
          [`ic-divider-${borderStyle}`]: true,
        }}
        {...((orientation === "vertical" ||
          (!!renderLabel() && !!isPropDefined(labelPlacement))) && {
          "aria-orientation": orientation,
          role: "separator",
        })}
      >
        {orientation === "horizontal" &&
          (!renderLabel() || !isPropDefined(labelPlacement)) && <hr />}
        {!!isPropDefined(labelPlacement) && !!renderLabel() && renderLabel()}
        {orientation === "vertical" &&
          (!renderLabel() || !isPropDefined(labelPlacement)) && (
            <div class="vertical-divider"></div>
          )}
      </Host>
    );
  }
}
