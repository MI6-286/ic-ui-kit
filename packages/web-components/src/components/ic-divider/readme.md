# ic-divider



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute         | Description                                                                                                                                                                                                                                                                              | Type                                                 | Default        |
| ---------------- | ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- | -------------- |
| `appearance`     | `appearance`      | The appearance of the divider.                                                                                                                                                                                                                                                           | `"dark" \| "default" \| "light" \| "theme"`          | `"default"`    |
| `borderStyle`    | `border-style`    | The line style of the divider.                                                                                                                                                                                                                                                           | `"dashed" \| "solid"`                                | `"solid"`      |
| `label`          | `label`           | The label for the divider.                                                                                                                                                                                                                                                               | `string`                                             | `undefined`    |
| `labelPlacement` | `label-placement` | The position the label is placed on the divider. `Left` and `right` placement is only applicable when orientation is set to `horizontal`. `Top` and `bottom` placement is only applicable when orientation is set to `vertical`. `Center` placement is applicable for both orientations. | `"bottom" \| "center" \| "left" \| "right" \| "top"` | `"center"`     |
| `orientation`    | `orientation`     | The orientation of the divider.                                                                                                                                                                                                                                                          | `"horizontal" \| "vertical"`                         | `"horizontal"` |
| `weight`         | `weight`          | The thickness of the divider.                                                                                                                                                                                                                                                            | `"medium" \| "thick" \| "thin" \| "very-thick"`      | `"thin"`       |


## CSS Custom Properties

| Name                            | Description                                                        |
| ------------------------------- | ------------------------------------------------------------------ |
| `--ic-divider-horizontal-width` | The width of a horizontal divider. The default value is `inherit`. |
| `--ic-divider-vertical-height`  | The height of a vertical divider. The default value is `inherit`.  |
| `--ic-divider-weight`           | The thickness of the divider.                                      |


## Dependencies

### Used by

 - [ic-side-navigation](../ic-side-navigation)

### Depends on

- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-divider --> ic-typography
  ic-side-navigation --> ic-divider
  style ic-divider fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


