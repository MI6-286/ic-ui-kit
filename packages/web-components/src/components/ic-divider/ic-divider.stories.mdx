import { Meta, Story, Canvas, Description } from "@storybook/addon-docs";
import { Markdown } from "@storybook/blocks";
import { html } from "lit-html";
import readme from "./readme.md?raw";

<Meta
  title="Web Components/Divider"
  component="ic-divider"
  parameters={{ loki: { skip: true } }}
/>

<Markdown>{readme}</Markdown>

### Appearance

<Canvas>
  <Story name="Appearance">
    {html` <div
      style="display: flex; flex-direction: column; gap: var(--ic-space-xs)"
    >
      <ic-divider appearance="default" weight="medium"></ic-divider>
      <ic-divider appearance="dark" weight="medium"></ic-divider>
      <div
        style="display:flex; align-items: center; background-color: #121212; height: var(--ic-space-md); width: 100%"
      >
        <ic-divider appearance="light" weight="medium"></ic-divider>
      </div>
      <ic-divider appearance="theme" weight="medium"></ic-divider>
    </div>`}
  </Story>
</Canvas>

### Weight

<Canvas>
  <Story name="Weight">
    {html` <div
      style="display: flex; flex-direction: column; gap: var(--ic-space-md)"
    >
      <ic-divider weight="thin"></ic-divider>
      <ic-divider weight="thin" border-style="dashed"></ic-divider>
      <ic-divider
        weight="thin"
        label="End of section"
        label-placement="center"
      ></ic-divider>
      <ic-divider weight="medium"></ic-divider>
      <ic-divider weight="medium" border-style="dashed"></ic-divider>
      <ic-divider
        weight="medium"
        label="End of section"
        label-placement="center"
      ></ic-divider>
      <ic-divider weight="thick"></ic-divider>
      <ic-divider weight="thick" border-style="dashed"></ic-divider>
      <ic-divider
        weight="thick"
        label="End of section"
        label-placement="center"
      ></ic-divider>
      <ic-divider weight="very-thick"></ic-divider>
      <ic-divider weight="very-thick" border-style="dashed"></ic-divider>
      <ic-divider
        weight="very-thick"
        label="End of section"
        label-placement="center"
      ></ic-divider>
    </div>`}
  </Story>
</Canvas>

### Border style

<Canvas>
  <Story name="Border style">
    {html` <div
      style="display: flex; flex-direction: column; gap: var(--ic-space-md)"
    >
      <ic-divider border-style="solid"></ic-divider>
      <ic-divider border-style="dashed"></ic-divider>
    </div>`}
  </Story>
</Canvas>

### Vertical

<Canvas>
  <Story name="Vertical">
    {html`
      <div style="display: flex; gap: var(--ic-space-lg); height: 400px">
        <ic-divider orientation="vertical" weight="thin"></ic-divider>
        <ic-divider
          orientation="vertical"
          weight="thin"
          border-style="dashed"
        ></ic-divider>
        <ic-divider orientation="vertical" weight="medium"></ic-divider>
        <ic-divider
          orientation="vertical"
          weight="medium"
          border-style="dashed"
        ></ic-divider>
        <ic-divider orientation="vertical" weight="thick"></ic-divider>
        <ic-divider
          orientation="vertical"
          weight="thick"
          border-style="dashed"
        ></ic-divider>
        <ic-divider orientation="vertical" weight="very-thick"></ic-divider>
        <ic-divider
          orientation="vertical"
          weight="very-thick"
          border-style="dashed"
        ></ic-divider>
      </div>
    `}
  </Story>
</Canvas>

### Label placement

<Canvas>
  <Story name="Label placement">
    {html` <div
        style="display: flex; flex-direction: column; gap: var(--ic-space-lg)"
      >
        <ic-divider label="Label left" label-placement="left"></ic-divider>
        <ic-divider label="Label center" label-placement="center"></ic-divider>
        <ic-divider label="Label right" label-placement="right"></ic-divider>
        <ic-divider label-placement="left">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label left</p>
          </ic-typography></ic-divider
        >
        <ic-divider label-placement="center">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label center</p>
          </ic-typography></ic-divider
        >
        <ic-divider label-placement="right">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label right</p>
          </ic-typography></ic-divider
        >
      </div>
      <div
        style="display: flex; gap: var(--ic-space-lg); height: 200px; margin-top: var(--ic-space-lg)"
      >
        <ic-divider
          label="Label top"
          label-placement="top"
          orientation="vertical"
        ></ic-divider>
        <ic-divider
          label="Label center"
          label-placement="center"
          orientation="vertical"
        ></ic-divider>
        <ic-divider
          label="Label bottom"
          label-placement="bottom"
          orientation="vertical"
        ></ic-divider>
        <ic-divider label-placement="top" orientation="vertical">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label top</p>
          </ic-typography></ic-divider
        >
        <ic-divider label-placement="center" orientation="vertical">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label center</p>
          </ic-typography></ic-divider
        >
        <ic-divider label-placement="bottom" orientation="vertical">
          <ic-typography variant="label" slot="label">
            <p style="width: max-content;">Slotted label bottom</p>
          </ic-typography></ic-divider
        >
      </div>`}
  </Story>
</Canvas>

### Playground

export const defaultArgs = {
  orientation: "horizontal",
  appearance: "default",
  "border-style": "solid",
  weight: "thin",
  label: "",
  "label-placement": "none",
  "--ic-divider-horizontal-width": "100%",
  "--ic-divider-vertical-height": "400px",
};

<Canvas>
  <Story
    args={defaultArgs}
    parameters={{ loki: { skip: true } }}
    argTypes={{
      orientation: {
        options: ["horizontal", "vertical"],
        control: {
          type: "inline-radio",
        },
      },
      appearance: {
        options: ["default", "dark", "light", "theme"],
        control: {
          type: "radio",
        },
      },
      "border-style": {
        options: ["solid", "dashed"],
        control: {
          type: "inline-radio",
        },
      },
      weight: {
        options: ["thin", "medium", "thick", "very-thick"],
        control: {
          type: "radio",
        },
      },
      label: {
        mapping: {
          none: "",
        },
      },
      "label-placement": {
        options: ["none", "left", "center", "right", "top", "bottom"],
        mapping: {
          none: "",
        },
        control: {
          type: "radio",
        },
      },
    }}
    name="Playground"
  >
    {(args) => {
      const backgroundColor =
        args.appearance === "light" ? "#121212" : "transparent";
      return html`<div
        style="display: flex; width: 100%; height: 100%; padding: var(--ic-space-md); background-color: ${backgroundColor}; box-sizing: border-box;"
      >
        <ic-divider
          orientation=${args.orientation}
          appearance=${args.appearance}
          border-style=${args["border-style"]}
          weight=${args.weight}
          label=${args.label}
          label-placement=${args["label-placement"]}
          style="--ic-divider-horizontal-width: ${args[
            "--ic-divider-horizontal-width"
          ]}; --ic-divider-vertical-height: ${args[
            "--ic-divider-vertical-height"
          ]}"
        ></ic-divider>
      </div>`;
    }}
  </Story>
</Canvas>
