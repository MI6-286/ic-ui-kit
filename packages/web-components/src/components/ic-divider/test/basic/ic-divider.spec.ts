import { newSpecPage } from "@stencil/core/testing";
import { Divider } from "../../ic-divider";

describe("ic-divider", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<ic-divider></ic-divider>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("correctly sets foregroundColor on theme change", async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<ic-side-navigation version="v0.0.0" status="BETA" app-title="ACME">
      <ic-navigation-item slot="primary-navigation" href="/" label="Home">
            <svg
              slot="icon"
              width="24"
              height="24"
              viewBox="0 0 24 24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 6.19L17 10.69V18.5H15V12.5H9V18.5H7V10.69L12 6.19ZM12 3.5L2 12.5H5V20.5H11V14.5H13V20.5H19V12.5H22L12 3.5Z"
                fill="currentColor"
              />
            </svg>
          </ic-navigation-item>
      <ic-divider slot="primary-navigation"></ic-divider>
        <ic-navigation-item slot="primary-navigation" href="/" label="a11y">
          <svg
            slot="icon"
            width="24"
            height="24"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C13.1 2 14 2.9 14 4C14 5.1 13.1 6 12 6C10.9 6 10 5.1 10 4C10 2.9 10.9 2 12 2ZM21 9H15V22H13V16H11V22H9V9H3V7H21V9Z"
              fill="currentColor"
            />
          </svg>
        </ic-navigation-item>
        </ic-side-navigation>`,
    });

    expect(page.rootInstance.foregroundColor).toEqual("light");

    expect(page.root).toMatchSnapshot("theme-change-light");

    await page.rootInstance.themeChangeHandler({
      detail: { mode: "dark" },
    });
    await page.waitForChanges();

    expect(page.rootInstance.foregroundColor).toEqual("dark");

    expect(page.root).toMatchSnapshot("theme-change-dark");
  });
});
