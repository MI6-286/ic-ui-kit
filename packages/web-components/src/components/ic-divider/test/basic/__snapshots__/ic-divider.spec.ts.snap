// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-divider correctly sets foregroundColor on theme change: theme-change-dark 1`] = `
<ic-divider aria-hidden="true" class="ic-divider-dark ic-divider-horizontal ic-divider-solid ic-divider-thin" slot="primary-navigation">
  <!---->
  <hr class="ic-divider-horizontal">
</ic-divider>
`;

exports[`ic-divider correctly sets foregroundColor on theme change: theme-change-light 1`] = `
<ic-divider aria-hidden="true" class="ic-divider-horizontal ic-divider-light ic-divider-solid ic-divider-thin" slot="primary-navigation">
  <!---->
  <hr class="ic-divider-horizontal">
</ic-divider>
`;

exports[`ic-divider should render 1`] = `
<ic-divider aria-hidden="true" class="ic-divider-default ic-divider-horizontal ic-divider-solid ic-divider-thin">
  <!---->
  <hr class="ic-divider-horizontal">
</ic-divider>
`;
